sound/soc/soc-pcm.c

```diff
@@ -2918,30 +2919,41 @@ int soc_new_pcm(struct snd_soc_pcm_runtime *rtd, int num)
 	}
 
 	if (rtd->dai_link->dynamic || rtd->dai_link->no_pcm) {
-		for_each_rtd_cpu_dais(rtd, i, cpu_dai) {
+		if (rtd->dai_link->dpcm_playback) {
+			invalid_dais = 0;
+			stream = SNDRV_PCM_STREAM_PLAYBACK;
+
+			for_each_rtd_cpu_dais(rtd, i, cpu_dai)
+				if (!snd_soc_dai_stream_valid(cpu_dai, stream))
+					invalid_dais++;
```

Comment:
Do you really need to know how many invalid DAIs there are? How about just
```
if (!snd_soc_dai_stream_valid(cpu_dai, stream)) {
	dev_err(rtd->card->dev,
			"invalid CPU DAI(s) detected for rtd %s without playback support\n",
			rtd->dai_link->stream_name);
	return -EINVAL;
}
playback = 1;
```

Reply:
i don't know. I feel saying 'it's broke' without providing additional information is not very useful.